app/services/rag_wise/chunking_service.rb

```diff
@@ -18,20 +18,34 @@ def file_chunking
              reader.pages.map(&:text).join(' ')
            when '.txt'
              File.read(@file.path)
+           when '.docx'
+             doc = Docx::Document.open(@file.path)
+             doc.paragraphs.map(&:text).join(' ')
+           when '.ipynb'
+             parse_ipynb(@file.path)
            else
              raise "Unsupported file type: #{@file_type}"
            end
 
     @text = text.gsub(/\s+/, ' ').strip
-    fixed_size_chunk_text(500, 100)
+    fixed_size_chunk_text(800, 100)
   end
 
   def text_chunking
-    fixed_size_chunk_text(500, 100)
+    fixed_size_chunk_text(800, 100)
   end
 
   private
 
+  def parse_ipynb(file_path)
+    notebook = JSON.parse(File.read(file_path))
+
+    notebook['cells'].
+      select { |cell| ['markdown', 'code', 'raw'].include?(cell['cell_type']) }.
+      map { |cell| cell['source'].join }.
+      join("\n")
+  end
+
   def fixed_size_chunk_text(chunk_size, overlap_size)
```

Comment:
Nit: why aren't you using `Langchain::Chunker::RecursiveText.new(text, 800, 100, ["\n\n", "\n", " ", ""]).chunks`?  
https://github.com/patterns-ai-core/langchainrb/blob/240dce06b0ee1cbc266aafde65c80b3b59af0394/lib/langchain/chunker/recursive_text.rb#L10

Instead of writing your own naive chunker, better to use a recursive chunker which would have better semantic separation.

Reply:
Good point, will use Langchain::Chunker::RecursiveText